### PR TITLE
SDIT-1713 Add AccessDeniedException (403) to POST /api/schedules/{agencyId}/count-activities

### PR DESCRIPTION
**Pull request by karenmillermoj**

### CODE REVIEW

File: src/main/java/uk/gov/justice/hmpps/prison/service/AgencyService.java [hunk 1]
```diff
@@ -225,6 +225,10 @@ public void verifyAgencyAccess(final String agencyId, boolean accessDeniedError,
                 logClientUnauthorisedAccess(agencyId);
                 throw new AccessDeniedException(format("Client not authorised to access agency with id %s due to missing override role%s", agencyId, allowInactive ? "" : ", or agency inactive"));
             }
+            if (accessDeniedError) {
```
Comment (karenmillermoj) [inline src/main/java/uk/gov/justice/hmpps/prison/service/AgencyService.java:228]: Added to match BoookingService

File: src/main/java/uk/gov/justice/hmpps/prison/service/AgencyService.java [hunk 2]
```diff
@@ -246,7 +250,9 @@ private void logClientUnauthorisedAccess(final String agencyId) {
     private void logUserUnauthorisedAccess(final String agencyId, final Set<String> agencyIds) {
         final Map<String, String> logMap = new HashMap<>();
         logMap.put("agencyId", agencyId);
```
Comment (karenmillermoj) [inline src/main/java/uk/gov/justice/hmpps/prison/service/AgencyService.java:252]: Added to match BoookingService

File: src/test/java/uk/gov/justice/hmpps/prison/api/resource/impl/ScheduleResourceTest.kt [hunk 3]
```diff
@@ -527,19 +587,6 @@ class ScheduleResourceTest : ResourceTest() {
       assertThat(response.statusCode.value()).isEqualTo(400)
     }
 
-    @Test
```
Comment (karenmillermoj) [inline src/test/java/uk/gov/justice/hmpps/prison/api/resource/impl/ScheduleResourceTest.kt:530]: test rewritten above

**Review by steverendell — APPROVED** [review]
lgtm

**Pull request merged**